packages/hebao_v1/contracts/modules2/core/SignedRequest.sol

```diff
@@ -0,0 +1,81 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.6.6;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/Controller.sol";
+
+import "../../lib/EIP712.sol";
+import "../../lib/SignatureUtil.sol";
+
+import "../../modules/security/GuardianUtils.sol";
+
+
+library SignedRequest {
+    using SignatureUtil for bytes32;
+
+    struct Request {
+        address[] signers;
+        bytes[]   signatures;
+        uint      nonce;
+        address   wallet;
+    }
+
+    string  public constant REQUEST_TYPE = "Request(address[] signers,bytes[] signatures,uint256 nonce,address wallet)";
+    bytes32 public constant REQUEST_TYPEHASH = keccak256(abi.encodePacked(REQUEST_TYPE));
+
+    function hash(Request memory request)
+        public
+        returns (bytes32)
+    {
+        return keccak256(
+            abi.encode(
+                REQUEST_TYPEHASH,
+                keccak256(abi.encode(request.signers)),
+                keccak256(abi.encode(request.signatures)),
```

Comment:
I don't think this can be a part of the data that is signed by all the signers, otherwise we end up with the same problem we had before where the signers array needs to be signed by all signers beforehand. And the signatures also being part of it makes it impossible I guess.

I don't think this is needed and they can just be removed here.

Reply:
removed